apps/ledger-live-mobile/src/newArch/features/Web3Hub/screens/Web3HubTabs/index.tsx

```diff
@@ -1,28 +1,115 @@
-import React from "react";
-import { View } from "react-native";
-import { Text } from "@ledgerhq/native-ui";
+import { Flex, IconsLegacy, Text } from "@ledgerhq/native-ui";
+import { useTheme } from "@react-navigation/native";
+import { FlashList } from "@shopify/flash-list";
+import type { TabData, TabsProps } from "LLM/features/Web3Hub/types";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { StyleSheet, TouchableOpacity, View } from "react-native";
 import { SafeAreaView } from "react-native-safe-area-context";
-import type { TabsProps } from "LLM/features/Web3Hub/types";
-import { ScreenName } from "~/const";
+import { NavigatorName, ScreenName } from "~/const";
 import Header from "./components/Header";
+import deviceStorage from "~/logic/storeWrapper";
+import TabItem, { Web3HubTabType } from "./components/TabItem";
 
 const edges = ["top", "bottom", "left", "right"] as const;
 
+const identityFn = (item: TabData) => item.id;
+
+const newTab = "New tab";
+
 export default function Web3HubTabs({ navigation }: TabsProps) {
+  const { colors } = useTheme();
+  const [tabs, setTabs] = useState<TabData[]>([]);
+
+  const listRef = useRef(null);
+
+  const handleItemClosePress = (itemId: string) => {
+    const filteredTabs = tabs.filter(item => item.id !== itemId);
+    deviceStorage.save("web3hub__TabHistory", filteredTabs);
+    setTabs(filteredTabs);
+  };
+  const goToSearch = useCallback(() => {
+    navigation.push(NavigatorName.Web3Hub, {
+      screen: ScreenName.Web3HubSearch,
+    });
+  }, [navigation]);
+  useEffect(() => {
+    const getTabs = async () => {
+      try {
+        const tabHistory =
+          ((await deviceStorage.get("web3hub__TabHistory")) as Web3HubTabType[]) || [];
+        setTabs(tabHistory);
+      } catch (error) {
+        console.error("Error fetching tabs from storage:", error);
+      }
+    };
+    getTabs();
+    const timer = setTimeout(() => {
+      getTabs();
+    }, 400);
+
+    return () => clearTimeout(timer);
+  }, []);
+
   return (
     <SafeAreaView edges={edges} style={{ flex: 1 }}>
       <Header
-        title={"N Tabs"} // Temporary, will probably be changed
+        title={`${tabs.length} ${tabs.length > 1 ? "tabs" : "tab"}`}
         navigation={navigation}
       />
 
-      <View
-        style={{
-          flex: 1,
-        }}
-      >
-        <Text>{ScreenName.Web3HubTabs}</Text>
+      <View style={styles.container}>
+        <FlashList
+          ref={listRef}
+          testID="web3hub-tabs-scroll"
+          keyExtractor={identityFn}
+          renderItem={({ item }) => (
+            <TabItem
+              item={item}
+              extraData={{ colors: colors }}
+              navigation={navigation}
+              onItemClosePress={handleItemClosePress}
+            />
+          )}
```

Comment:
As discussed previously, you don't want to use declare the `renderItem` function in the render function, it should be declared outside to avoid using variable in the render scope directly as you do with `navigation` and `onItemClosePress`

Instead we should copy the way it's done in other parts of web3hub, using the `extraData` field in the `FlashList` to pass data to `renderItem` and then `TabItem`

Reply:
[9ab981a1da7456c651a2935e96737c600f045876](https://github.com/LedgerHQ/ledger-live/pull/7962/commits/9ab981a1da7456c651a2935e96737c600f045876?diff=unified&w=1#diff-4144aa8182c6bd3d7895e6805d0466c38aea8e3bed5d00da11ace22fbad68480L65-R84)